tests/pipelines/test_pipelines_text_to_audio.py

```diff
@@ -0,0 +1,242 @@
+            ANY(np.ndarray),
+            audio,
+        )
+
+        # test two examples side-by-side
+        outputs = speech_generator(["This is a test", "This is a second test"], forward_params=forward_params)
+
+        audio = [output["audio"] for output in outputs]
+
+        self.assertEqual(
+            [
+                ANY(np.ndarray),
+                ANY(np.ndarray),
+            ],
+            audio,
+        )
+
+        # test batching
+        outputs = speech_generator(
+            ["This is a test", "This is a second test"], forward_params=forward_params, batch_size=2
+        )
+
+        self.assertEqual(
+            ANY(np.ndarray),
+            outputs[0]["audio"],
+        )
+
+    @slow
+    @require_torch
+    def test_large_model_pt(self):
+        speech_generator = pipeline(task="text-to-audio", model="suno/bark-small", framework="pt")
+
+        # test text-to-speech
+
+        forward_params = {
+            # Using `do_sample=False` to force deterministic output
+            "do_sample": False,
+            "semantic_max_new_tokens": 100,
+        }
+
+        # atm, must do to stay coherent with BarkProcessor
+        preprocess_params = {
+            "max_length": 256,
+            "add_special_tokens": False,
+            "return_attention_mask": True,
+            "return_token_type_ids": False,
+            "padding": "max_length",
+        }
+
+        outputs = speech_generator(
+            "This is a test", forward_params=forward_params, preprocess_params=preprocess_params
+        )
+
+        self.assertEqual(
+            {"audio": ANY(np.ndarray), "sampling_rate": 24000},
+            outputs,
+        )
+
+        # test two examples side-by-side
+        outputs = speech_generator(
+            ["This is a test", "This is a second test"],
+            forward_params=forward_params,
+            preprocess_params=preprocess_params,
+        )
+
+        audio = [output["audio"] for output in outputs]
+
+        self.assertEqual(
+            [
+                ANY(np.ndarray),
+                ANY(np.ndarray),
+            ],
+            audio,
+        )
+
+        # test other generation strategy
+
+        forward_params = {
+            "do_sample": True,
+            "semantic_max_new_tokens": 100,
+            "semantic_num_return_sequences": 2,
+        }
+
+        outputs = speech_generator(
+            "This is a test",
+            forward_params=forward_params,
+            preprocess_params=preprocess_params,
+        )
+
+        audio = outputs["audio"]
+
+        self.assertEqual(
+            ANY(np.ndarray),
+            audio,
+        )
+
+        # test using a speaker embedding
+
+        processor = AutoProcessor.from_pretrained("suno/bark-small")
+
+        temp_inp = processor("hey, how are you?", voice_preset="v2/en_speaker_5")
+
+        history_prompt = temp_inp["history_prompt"]
+
+        forward_params["history_prompt"] = history_prompt
```

Comment:
Another limitation of the current TTS pipeline is the way a speaker embeddings, or an additional conditional input parameter must be passed.

Here, to use Bark speaker embeddings, one must do a blank call to `BarkProcessor`.